### PR TITLE
Milestone-Eight: Integration spec and 

### PR DESCRIPTION
**Pull request by saluumaa**
In this Milestone I:

1. Solve N+1 problem 
2. Add the integration test for views

    - Users index and show page
    - Posts index and show page

3. This is the pictures before and after N+1 
After n+1 intergration test
before N+1 integration

### CODE REVIEW

File: spec/views/users/index.html.erb_spec.rb [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+  end
+  it 'displays the users profile picture' do
+    expect(page).to have_selector("img[src='#{user.photo}']")
+  end
+  it 'displays the users name' do
+    expect(page).to have_content(user.name)
+  end
+  it 'displays the users show page when I click on the users name' do
+    click_link user.name
+    expect(page).to have_current_path(user_path(user))
```
Comment (rloterh) [inline spec/views/users/index.html.erb_spec.rb]: - [ ] A test to assert that the scenario **_I can see the number of posts each user has written_** for _user index view_ spec is missing! Please ensure to improve upon this aspect.

File: spec/views/users/show.html.erb_spec.rb [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+    expect(page).to have_content(post3.title)
+    expect(page).to have_content(post4.title)
+  end
+  it 'displays the see all button' do
+    expect(page).to have_link('See all posts')
+  end
+  it 'displays the posts index page when click see all posts' do
+    click_link 'See all posts'
+    expect(current_path).to eq "#{user_posts_path(user)}/"
+  end
```
Comment (rloterh) [inline spec/views/users/show.html.erb_spec.rb]: - [ ] Similarly, a test to assert that the scenario **_When I click a user's post, it redirects me to that post's show page_** for _user show view_ spec is missing! Please ensure to improve upon this aspect.

File: spec/views/posts/index.html.erb_spec.rb [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+  it 'displays the number of likes the post has' do
+    expect(page).to have_content(post.likes_counter)
+  end
+
+  it 'displays the pagination button' do
+    expect(page).to have_button('pagination')
+  end
+
+  it 'displays post show page when I click on the post title' do
+    click_link post.title
```
Comment (rloterh) [inline spec/views/posts/index.html.erb_spec.rb]: - [ ] Again, a test to assert that the scenario **_I can see some of the post's body_** for _post index view_ spec is missing! Please ensure to improve upon this aspect.

File: spec/views/posts/show.html.erb_spec.rb [hunk 4]
```diff
@@ -1,5 +1,39 @@
...
+    expect(page).to have_content(post.text)
+  end
+
+  it 'displays the commentor username' do
+    expect(page).to have_content(comment.author.name)
+  end
+
+  it 'displays the comment' do
+    expect(page).to have_content(comment.text)
+  end
```
Comment (rloterh) [inline spec/views/posts/show.html.erb_spec.rb]: - [ ] A number of _post view_ specs are failing. Let's ensure all test cases are passing!

**Failing spec cases:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Salma, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🏁 🚦  ⛪ 🚜  🏃‍♀️  🌴  🏫 🌳 🏘️ 

## Highlights 👍  
- N + 1 problem addressed!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Salma, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🏁 🚦  ⛪ 🚜  🏃‍♀️  🌴  🏫 🌳 🏘️ 

## Highlights 👍  
- N + 1 problem addressed!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @saluumaa , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Well done adding the remaining tests cases. ✔
- All the tests are now passing. ✔
- Now the milestone requirements have been met with success. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**